src/bindings/c/include/openvino/c/ov_node.h

```diff
@@ -35,70 +44,93 @@ typedef struct {
-ov_node_list_get_shape_by_index(ov_output_node_list_t* nodes, size_t idx, ov_shape_t* shape);
+ov_node_list_get_shape_by_index(ov_output_const_node_list_t* port_list, size_t idx, ov_shape_t* shape);
 
 /**
- * @brief Get the partial shape of ov_output_node.
+ * @brief Get the partial shape of port with index in port_list.
  * @ingroup node
- * @param nodes A pointer to the ov_output_node_list_t.
- * @param idx Index of the input tensor
- * @param tensor_shape tensor shape.
+ * @param port_list A pointer to the ov_output_const_node_list_t.
+ * @param idx Index in the port_list.
+ * @param partial_shape Partial shape.
  * @return Status code of the operation: OK(0) for success.
  */
 OPENVINO_C_API(ov_status_e)
-ov_node_list_get_partial_shape_by_index(ov_output_node_list_t* nodes, size_t idx, ov_partial_shape_t** partial_shape);
+ov_node_list_get_partial_shape_by_index(ov_output_const_node_list_t* port_list, size_t idx, ov_partial_shape_t** partial_shape);
 
 /**
- * @brief Get the tensor type of ov_output_node.
+ * @brief Get the tensor type of port by index.
  * @ingroup node
- * @param nodes A pointer to the ov_output_node_list_t.
- * @param idx Index of the input tensor
+ * @param nodes A pointer to the ov_output_const_node_list_t.
+ * @param idx Index in the port_list.
  * @param tensor_type tensor type.
  * @return Status code of the operation: OK(0) for success.
  */
 OPENVINO_C_API(ov_status_e)
-ov_node_list_get_element_type_by_index(ov_output_node_list_t* nodes, size_t idx, ov_element_type_e* tensor_type);
+ov_node_list_get_element_type_by_index(ov_output_const_node_list_t* port_list, size_t idx, ov_element_type_e* tensor_type);
+
+/**
+ * @brief free port_list
+ * @ingroup node
+ * @param port_list The pointer to the instance of the ov_output_node_list_t to free.
+ */
+OPENVINO_C_API(void) ov_output_node_list_free(ov_output_node_list_t* port_list);
+
+/**
+ * @brief free const port_list
+ * @ingroup node
+ * @param port_list The pointer to the instance of the ov_output_const_node_list_t to free.
+ */
+OPENVINO_C_API(void) ov_output_const_node_list_free(ov_output_const_node_list_t* port_list);
 
 /**
- * @brief free ov_output_node_list_t
+ * @brief free port object
  * @ingroup node
- * @param output_nodes The pointer to the instance of the ov_output_node_list_t to free.
+ * @param port The pointer to the instance of the ov_output_node_t to free.
  */
-OPENVINO_C_API(void) ov_output_node_list_free(ov_output_node_list_t* output_nodes);
+OPENVINO_C_API(void) ov_output_node_free(ov_output_node_t* port);
 
 /**
- * @brief free ov_output_const_node_t
+ * @brief free const port
  * @ingroup node
- * @param output_node The pointer to the instance of the ov_output_const_node_t to free.
+ * @param port The pointer to the instance of the ov_output_const_node_t to free.
  */
-OPENVINO_C_API(void) ov_output_node_free(ov_output_const_node_t* output_node);
+OPENVINO_C_API(void) ov_output_const_node_free(ov_output_const_node_t* port);
```

Comment:
why do we need const and non-const nodes? I suppose in C API they all have const semantic

Reply:
In C++, two kinds of nodes are defined, one is const and the other is non-const. They respectively call different functions.